### DISCUSSION TITLE
Any reasonable way to access the HTTP/1 response reason string?

### DISCUSSION THREAD

**Discussion by uberjay** [socketry/async-http#145] [Q&A] [answered]
After a lot of confusion, I finally determined that the [Proxmox VE API](https://pve.proxmox.com/wiki/Proxmox_VE_API) delivers its error reasons for failed requests using the HTTP response reason string.

Because it's not obviously documented anywhere, and because async-http discards it:

```
	class Response < Protocol::Response
	  # @param reason [String] HTTP response line reason, ignored.
	  def initialize(connection, version, status, reason, headers, body)
	    @connection = connection
	    
	    protocol = headers.delete(UPGRADE)
	    
	    super(version, status, headers, body, protocol)
	  end
```

It took me _far_ to long to figure this out. Short of monkey-patching the HTTP1::Response class 🤢, is there a less disgusting way for me to get this information? Perhaps a way to specify what response class should be used?

Any help is appreciated!

This works, but I don't like it:

```
module Async
  module HTTP
    module Protocol
      module HTTP1
	class Response < Protocol::Response
          attr_reader :reason

	  # @param reason [String] HTTP response line reason, ignored.
	  def initialize(connection, version, status, reason, headers, body)
	    @connection = connection
            @reason = reason
	    
	    protocol = headers.delete(UPGRADE)
	    
	    super(version, status, headers, body, protocol)
	  end
	end
      end
    end
  end
end
```

**Comment by ioquatix** [comment]
I am okay with your suggested change, do you want to make a PR?

**Comment by uberjay** [comment]
PR: https://github.com/socketry/async-http/pull/146

**Reply by ioquatix** [reply, marked as answer]
For the record, I've merged it, and will release it this week.
